splice_machine/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/GroupByCorrelationIT.java

```diff
@@ -0,0 +1,165 @@
+            Assert.assertTrue(resultSet.next());
+            Assert.assertEquals("E2", resultSet.getString(1));Assert.assertEquals("P1", resultSet.getString(2));Assert.assertEquals(4, resultSet.getInt(3));
+            Assert.assertTrue(resultSet.next());
+            Assert.assertEquals("E2", resultSet.getString(1));Assert.assertEquals("P2", resultSet.getString(2));Assert.assertEquals(4, resultSet.getInt(3));
+            Assert.assertTrue(resultSet.next());
+            Assert.assertEquals("E3", resultSet.getString(1));Assert.assertEquals("P2", resultSet.getString(2));Assert.assertEquals(5, resultSet.getInt(3));
+            Assert.assertTrue(resultSet.next());
+            Assert.assertEquals("E4", resultSet.getString(1));Assert.assertEquals("P2", resultSet.getString(2));resultSet.getInt(3); Assert.assertTrue(resultSet.wasNull());
+            Assert.assertTrue(resultSet.next());
+            Assert.assertEquals("E4", resultSet.getString(1));Assert.assertEquals("P4", resultSet.getString(2));resultSet.getInt(3); Assert.assertTrue(resultSet.wasNull());
+            Assert.assertTrue(resultSet.next());
+            Assert.assertEquals("E4", resultSet.getString(1));Assert.assertEquals("P5", resultSet.getString(2));resultSet.getInt(3); Assert.assertTrue(resultSet.wasNull());
+            Assert.assertFalse(resultSet.next());
+        }
+    }
+
+    @Test
+    public void ssqWithoutCorrelationOnGroupByColumnThrows() throws Exception {
+        try(ResultSet resultSet = methodWatcher.executeQuery(String.format("SELECT oc1, (SELECT MAX(ic1) FROM %s WHERE ic2 = %s.oc2) FROM %s GROUP BY oc1 order by oc1 asc",
+                                                             itWatcher.toString(), otWatcher.toString(), otWatcher.toString()))) {
+            Assert.fail("expected exception containing message: The SELECT list of a grouped query contains at least one invalid expression. " +
+                                "If a SELECT list has a GROUP BY, the list may only contain valid grouping expressions and valid aggregate expressions.");
+        } catch(Exception se) {
+            Assert.assertTrue(se instanceof SQLException);
+            Assert.assertEquals("42Y30", ((SQLException)se).getSQLState());
+            Assert.assertTrue(se.getMessage().contains("The SELECT list of a grouped query contains at least one invalid expression. If a SELECT list has a GROUP BY, " +
+                                                               "the list may only contain valid grouping expressions and valid aggregate expressions."));
+        }
+    }
+
+    @Test
+    public void subqueryWithCorrelationOnGroupByColumnThrows() throws Exception {
+        try(ResultSet resultSet = methodWatcher.executeQuery(String.format("SELECT oc1, (SELECT c2 FROM %s WHERE ic2 = %s.oc1) FROM %s GROUP BY oc1 order by oc1 asc",
```

Comment:
I think `c2` should be `ic2`, this way, we should expect the error message `ERROR 21000: Scalar subquery is only allowed to return a single row.` at runtime instead of `c2` not found. Thanks!

Reply:
fixed it.